src/views/Generator/Allowlist/AllowlistDialog.tsx

```diff
@@ -66,6 +76,10 @@ export function AllowlistDialog({
     setIncludeStaticAssets(checked && staticAssetCount > 0)
   }
 
+  function shouldDisableSelectAll() {
+    return every(firstPartyFilteredHosts, (host) => includes(allowlist, host))
+  }
```

Comment:
NIT: This would be calculated on every render, maybe we could call this `isSelectAllDisabled` and wrap in `useMemo` instead?